### DISCUSSION TITLE
python, fastAPI, pydantic

### DISCUSSION THREAD

**Discussion by dgutson** [hofstadter-io/hof#105] [Ideas]
This seems to be the exact project I was looking for, so maybe we will start contributing here a lot.
Question: I would like to use hof to generate fastAPI in general and pydantic in particular.
Would this be aligned? We could contribute here.

**Comment by verdverm** [comment]
Hi @dgutson 

You can definitely do that with `hof` and without needing to modify `hof`'s code!

Are you trying to make a single API server, or a module that many can use to create APIs in Python? The difference is actually quite small, so if you change your mind later, the change is quite small or not even needed in many cases. You can contribute your work by pushing it to GitHub (or any git) and letting people know. We can even get it listed in the module library on the website so others can find it more easily.

What you will want to do is create a generator that has specifics to python, fastAPI, and pydantic. You can follow along the content at these links to get started:

- https://docs.hofstadter.io/first-example/
- https://github.com/hofstadter-io/demos/tree/main/adhoc-to-module

I don't have any python specific examples yet, but I recently added the `black` formatter in preparation, so it seems you have good timing!

Feel free to ask any follow up questions here or on slack. Happy to help you work through any unclear parts or when you get stuck.
